### PR TITLE
Temporarily clean s390x build workspace after build to save space

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Signed-off-by: Andrew Leonard <anleonar@redhat.com>

### CODE REVIEW

**Review by gdams — APPROVED** [review]

**Comment by andrew-m-leonard** [comment]
Test here: https://ci.adoptopenjdk.net/view/work-in-progress/job/andrew-jdk11u-linux-s390x-hotspot/1/console

**Comment by AdamBrousseau** [comment]
Related #2296

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -983,7 +983,7 @@ class Build {
                     throw new Exception()
                 }
             } finally {
-                if (buildConfig.TARGET_OS == "aix") {
+                if (buildConfig.TARGET_OS == "aix" || buildConfig.ARCHITECTURE == "s390x") {
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:986]: We should add a comment as to why we need to do this.

**Review by karianna — CHANGES_REQUESTED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 1 above)
Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy:989]: added comment

**Review by karianna — APPROVED** [review]

**Pull request merged**